app/src/components/TableComponent/TableComponent.tsx

```diff
@@ -0,0 +1,90 @@
+import { FlatList, StyleSheet, View } from "react-native";
+import React from "react";
+import NormalText from "../NormalText/NormalText";
+import { Heading5 } from "../NormalText/FontTypes";
+
+interface TableProps {
+  numColumns: number;
+  numRows: number;
+  tableHeader: string[];
+  tableData: any[][];
+  headerBackgroundColor?: string;
+  headerTextColor?: string;
+}
+
+const TableComponent = (props: TableProps) => {
+  const renderItem = (item: { item: string[]; index: number }) => {
+    const { item: rowData, index } = item;
+    return (
+      <View style={styles.row}>
+        {rowData.map((cellData, cellIndex) => (
+          <View key={cellIndex} style={styles.cell}>
+            <NormalText normalText={cellData} />
+          </View>
+        ))}
+      </View>
+    );
+  };
+
+  const header = props.tableHeader.map((title, index) => (
```

Comment:
Render the title as:
`.map((h, index) => <View
      key={index}
      style={[
        styles.headerCell,
        { backgroundColor: props.headerBackgroundColor },
      ]}
    >
      <NormalText
        normalText={r.title}
        textColor={props.headerTextColor}
        fontType={Heading5}
      />
    </View>)`